pipelines/README.md

```diff
@@ -1,45 +1,134 @@
+```python
+import torch
+from registry import get_model
+
+def toy_validation(config):
+  model = get_model(config["model"]).eval()
+  inp = torch.ones((1, 3, 32, 32)).float()
+  output = model(inp.to(config["device"]))
 ```
+[comment]:pipeline-end
 
-## How to use my own implementation of loss, model, augmentations, etc?
-You should put your python object inside the corresponding registry by some key.
-It allows you to access this object in the config file by that key.
+`registry.py` (maps entity's config to a Python constructor, **part of OML package**):
 
-You may change the following blocks and to work correctly some of them have to inherit our interfaces:
-* `Transforms`, `Sampler`, `Optimizer`, `Scheduler` - follow the standard PyTorch interfaces.
-* `Model` - have to be successor of `IExtractor` (see `oml.interfaces.models`)
-* `Criterion` - have to be successor of `ITripletLossWithMiner` (see `oml.interfaces.criterions`)
-  * You may want to change only `Miner` inside the criterion. It has to be a successor of `ITripletsMiner` (see `oml.interfaces.miners`).
+[comment]:registry-start
+```python
+from torchvision.models import resnet18, resnet50
 
+MODELS_REGISTRY = {"resnet18": resnet18, "resnet50": resnet50}
 
-Let's consider an example of using custom augmentations & model.
-Your `config.yaml` and `train.py` may look like this:
-```yaml
-...
+def get_model(config):
+  return MODELS_REGISTRY[config["name"]](**config["args"])
+```
+[comment]:registry-end
 
-transforms_train:
-  name: custom_augmentations
-  args: {}
+`config.yaml` (describes the whole run, **your local file**):
 
+[comment]:config-start
+```yaml
 model:
-  name: custom_model
+  name: resnet50
   args:
-    pretrained: True
+    weights: IMAGENET1K_V1
 
-...
+device: cpu
 ```
+[comment]:config-end
+
+`validate.py` (script which simply runs pipeline, **your local file**):
 
+[comment]:script-start
+```python
+import hydra
+from pipeline import toy_validation
+
+@hydra.main(config_name="config.yaml")
+def main_hydra(cfg):
+    toy_validation(cfg)
+
+if __name__ == "__main__":
+    main_hydra()
+```
+[comment]:script-end
+
+Shell command:
+
+[comment]:shell-start
+```shell
+
+python run.py model.args.weights=null
+
+```
+[comment]:shell-end
+
+Note, we use [Hydra](https://hydra.cc/docs/intro/) as a config parser. One of its abilities
+is to change part of the config from a command line, as showed above.
+
+
+## How to work with Pipelines?
+
+The recommended way is the following:
+1. Install OML: `pip install open-metric-learning`
+2. Prepare your dataset in the required [format](https://open-metric-learning.readthedocs.io/en/latest/oml/data.html). (There are [converters](https://github.com/OML-Team/open-metric-learning/tree/main/pipelines/datasets_converters) for 4 popular datasets).
+3. Go to Pipeline's folder and copy `.py` script and its `.yaml` to your workdir. Modify the config if needed.
+4. Run the script via the command line.
+
+## Building blocks of Pipelines
```

Comment:
I would move this section before `Minimal example of a Pipeline`.

Reply:
I cannot agree here, because there is a transition from `Building blocks of Pipelines`  to `How to use my own implementation of loss, model, etc.?`. If I break it with a `Minimal example of a Pipeline` it would break the narrative.